### PR TITLE
bump jest, ts-jest versions

### PR DESCRIPTION
**Pull request by brendarearden**
Addresses [#2319](https://github.com/stoplightio/prism/issues/2319)

**Summary**

Bumped ts-jest (has a direct semver dependency)
Bumped jest (ts-jest requires 29.x version)
Bumped @types/jest (bumped to match jest)

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [x] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [x] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [x] N/A

### CODE REVIEW

**Review by chohmann — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const { pathsToModuleNameMapper } = require('ts-jest/utils');
+const { pathsToModuleNameMapper } = require('ts-jest');
```
Comment (brendarearden) [inline jest.config.js:1]: https://kulshekhar.github.io/ts-jest/docs/getting-started/paths-mapping/ newest way to importpathsToModulepathsToModule

File: jest.config.js [hunk 2]
```diff
@@ -23,40 +23,52 @@ module.exports = {
       ...projectDefault,
       displayName: 'HTTP-SERVER',
       testMatch: ['<rootDir>/packages/http-server/src/**/__tests__/*.*.ts'],
-      globals: {
-        'ts-jest': {
-          tsconfig: '<rootDir>/packages/tsconfig.test.json',
-        },
+      transform: {
```
Comment (brendarearden) [inline jest.config.js:26]: when attempting to run tests I was getting 
```
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
```
so I moved the global settings for ts-jest into the transform section as recommended

File: packages/cli/src/commands/mock.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ const mockCommand: CommandModule = {
       jsonSchemaFakerFillProperties,
     };
 
-    return runPrismAndSetupWatcher(createPrism, options);
+    return runPrismAndSetupWatcher(createPrism, options) as Promise<void>;
```
Comment (brendarearden) [inline packages/cli/src/commands/mock.ts:47]: Type error was - I am sure there is a better way to solve this
```
Type '(parsedArgs: ArgumentsCamelCase<{}>) => Promise<unknown>' is not assignable to type '(args: ArgumentsCamelCase<{}>) => void | Promise<void>'.
      Type 'Promise<unknown>' is not assignable to type 'void | Promise<void>'.
        Type 'Promise<unknown>' is not assignable to type 'Promise<void>'.
          Type 'unknown' is not assignable to type 'void'.

```

File: yarn.lock [hunk 4]
```diff
@@ -7650,18 +6831,18 @@ semver@7.5.3:
   dependencies:
     lru-cache "^6.0.0"
 
-semver@7.x, semver@^7.0.0, semver@^7.1.1, semver@^7.2.1, semver@^7.3.2, semver@^7.3.4, semver@^7.3.5, semver@^7.3.7, semver@^7.3.8:
+semver@^6.0.0, semver@^6.3.0, semver@^6.3.1:
```
Comment (brendarearden) [inline yarn.lock:6834]: these are due to Babel with is a jest dependency. Babel has stated that their use of semver does not allow the vulnerability to be exploited so we can ignore the security warning.

**Comment by brendarearden** [comment]
actually - I was looking at this again - I am going to close out the issue and leave the PR.

This isn't something we need to do right now since our ts-jest dependencyin `master` is pointing to the latest semver

https://github.com/stoplightio/prism/blob/59de04269e22fbc99775462e19119d2017977588/yarn.lock#L8248

https://github.com/stoplightio/prism/blob/59de04269e22fbc99775462e19119d2017977588/yarn.lock#L7653

File: packages/cli/src/commands/mock.ts (hunk 3 above)
Comment (daniel-white) [reply packages/cli/src/commands/mock.ts:47]: is this indicative that the caller of this function won't get what it needs? or it just doesn't care?

Comment (brendarearden) [reply packages/cli/src/commands/mock.ts]: thanks for walking through the types changes with me, you should see our changes reflected in the most recent commit.

File: jest.config.js [hunk 5]
```diff
@@ -1,19 +1,11 @@
-const { pathsToModuleNameMapper } = require('ts-jest/utils');
+const { pathsToModuleNameMapper } = require('ts-jest');
 const path = require('path');
 const { mapValues } = require('lodash');
 const { compilerOptions } = require('./packages/tsconfig.test');
 
 const projectDefault = {
   moduleNameMapper: {
     ...mapValues(pathsToModuleNameMapper(compilerOptions.paths), v => path.resolve(path.join('packages', v))),
-
```
Comment (brendarearden) [inline jest.config.js:9]: this was added recently, but is no longer needed with the newer version of jest

Comment (chohmann) [reply jest.config.js:9]: This is awesome!

**Review by chohmann — APPROVED** [review]
Love it! So glad we can cleanup that weird tsconfig stuff

**Review by daniel-white — APPROVED** [review]

**Pull request merged**